### ISSUE TITLE
RuntimeException while trying to printfield values

### ISSUE THREAD

**Issue by AbhishekPuligundla** [jbachorik/btrace#179]
The source file is :+1: 

import java.io.*;
import java.util.HashMap;
class FirstClass
{
	String name;
	int id;
	FirstClass(String name, int value)
	{
		this.name=name;
		this.id=value;
	}
}
public class SecondClass
{
	HashMap<String, HashMap<Integer,FirstClass>> map1;
	public static void main(String args[])
	{
		
		try{ Thread.sleep(20000); }catch(Exception e){}
		SecondClass object1 =new SecondClass();
	}
	SecondClass()
	{
		map1 =new HashMap<String, HashMap<Integer,FirstClass>>();
		HashMap<Integer,FirstClass> map2 = new HashMap<Integer,FirstClass>();
		map2.put(1,new FirstClass("a",2));
		try{ Thread.sleep(5000); }catch(Exception e){}
		map2.put(2,new FirstClass("b",3));
		map2.put(3,new FirstClass("c",4));
		map1.put("Hello",map2);
		
	}
}

The script file is : 

import com.sun.btrace.annotations.*;
import static com.sun.btrace.BTraceUtils.*;
import com.sun.btrace.*;
import java.util.*;
@BTrace
public class TraceClass
{
	@OnMethod(clazz="SecondClass",method="/.*/",location=@Location(value=Kind.NEW,clazz="/.*/",where=Where.AFTER ))
	public static void traceMethod123456(@Self Object obj,String instanceclassname)
	{
		println("New Class");
		println(classOf(obj));
		println(instanceclassname);
		Class cl=classForName(instanceclassname);
		printFields(classOf(cl));
	}
}

The result is : 

New Class
class SecondClass
java.util.HashMap
{cachedConstructor=null, newInstanceCallerCache=null, name=java.lang.Class, refl
ectionData=java.lang.ref.SoftReference@5c647e05, classRedefinedCount=0, genericI
nfo=null, enumConstants=null, enumConstantDirectory=null, annotationData=java.la
ng.Class$AnnotationData@33909752, annotationType=null, classValueMap=null, }
New Class
class SecondClass
java.util.HashMap
{cachedConstructor=null, newInstanceCallerCache=null, name=java.lang.Class, refl
ectionData=java.lang.ref.SoftReference@5c647e05, classRedefinedCount=0, genericI
nfo=null, enumConstants=null, enumConstantDirectory=null, annotationData=java.la
ng.Class$AnnotationData@33909752, annotationType=null, classValueMap=null, }
New Class
class SecondClass
FirstClass
java.lang.RuntimeException: java.lang.ClassNotFoundException: FirstClass
        at com.sun.btrace.BTraceUtils.translate(BTraceUtils.java:6561)
        at com.sun.btrace.BTraceUtils.access$300(BTraceUtils.java:61)
        at com.sun.btrace.BTraceUtils$Reflective.classForName(BTraceUtils.java:5
316)
        at com.sun.btrace.BTraceUtils$Reflective.classForName(BTraceUtils.java:5
305)
        at com.sun.btrace.BTraceUtils.classForName(BTraceUtils.java:456)
        at SecondClass.$btrace$TraceClass$traceMethod123456(SecondClass.java)
        at SecondClass.<init>(SecondClass.java:26)
        at SecondClass.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: FirstClass
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at com.sun.btrace.BTraceUtils$Reflective.classForName(BTraceUtils.java:5
314)
        ... 5 more

how can i print the object and field values of the new class while using Kind.NEW

**Comment by jbachorik** [comment]
Try
```
@OnMethod(clazz="SecondClass",method="/./",location=@Location(value=Kind.NEW,clazz="/./",where=Where.AFTER ))
public static void traceMethod123456(@Self Object obj, @TargetInstance newInstance, String instanceclassname) {
    println("New Class: " + instanceclassname);
    printFields(newInstance, true);
}
```

If I understand correctly what you are trying to achieve this should do it.

**Comment by AbhishekPuligundla** [comment]
wasn't TargetInstance was supposed to be used when we are using Kind.CALL and also in the above code you didn't specify the identifier after @TargetInstance. The above code is showin the error

<Identifier> expected
BTrace compilation failed.

if I use the identifier it is showing the exception

com.sun.btrace.VerifierException: @CalledInstance annotation applicable only for
 Kind.CALL: traceMethod123456(Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Str
ing;)V(1)
        at com.sun.btrace.runtime.Verifier.reportError(Verifier.java:240)
        at com.sun.btrace.agent.Client.verifySpecialParameters(Client.java:450)
        at com.sun.btrace.agent.Client.verify(Client.java:419)
        at com.sun.btrace.agent.Client.loadClass(Client.java:236)
        at com.sun.btrace.agent.RemoteClient.<init>(RemoteClient.java:59)
        at com.sun.btrace.agent.Main.startServer(Main.java:405)
        at com.sun.btrace.agent.Main.access$000(Main.java:65)
        at com.sun.btrace.agent.Main$3.run(Main.java:168)
        at java.lang.Thread.run(Thread.java:745)


So which means the TargetInstance and TargetMethodOrField must be used when Kind.CALL must be used, right.

**Comment by jbachorik** [comment]
Sorry, my bad, should have been
```
@OnMethod(clazz="SecondClass",method="/./",location=@Location(value=Kind.NEW,clazz="/./",where=Where.AFTER ))
public static void traceMethod123456(@Self Object obj, @Return Object newInstance, String instanceclassname) {
    println("New Class: " + instanceclassname);
    printFields(newInstance, true);
}
```

I should enhance the location Kind enumeration with javadocs with allowed parameter annotations and their meaning in the context of each location. Also, the verifier outputs a confusing message using obsolete annotation name and obsolete location restrictions. All of this will need to be fixed.

**Comment by AbhishekPuligundla** [comment]
It is not printing anything.

**Comment by jbachorik** [comment]
It turns out that java verifier does not like passing around an instance without calling its constructor first.

**Comment by jbachorik** [comment]
Javadocs improvements - 16bf8a0
Verifier messages fix      - e3aee8d
